### PR TITLE
Contribute enhancements for render terminator to append

### PR DESCRIPTION
**Pull request by timlevett**
This adds the ability for a terminator redirect to append an fname of an application.

### CODE REVIEW

**Comment by drewwills** [comment]
### General Note on New Pull Requests

Please hold off on merging pull requests to `master` for a brief period.  A significant refactor is already underway for uPortal 5, and the potential for difficult conflicts is large.  I'll do my best to make this change happen quickly.  Thank you for your patience.

File: uportal-war/src/main/java/org/jasig/portal/rendering/RedirectRenderingPipelineTerminator.java [hunk 1]
```diff
@@ -53,10 +65,23 @@ public void renderState(final HttpServletRequest request, final HttpServletRespo
...
-        
+        if(StringUtils.isNotBlank(appender)) {
+          String thingToAppend = "";
+          if(APPENDER_FNAME.equalsIgnoreCase(appender)) {
+            try {
+              final IPortletDefinition portletDefinition = utils.getPortletDefinitionFromServletRequest(request);
+              thingToAppend = portletDefinition.getFName();
+            }
+            catch(Exception e){
+              logger.error("Appender was null but couldn't figure out fname.",e);
```
Comment (drewwills) [inline uportal-war/src/main/java/org/jasig/portal/rendering/RedirectRenderingPipelineTerminator.java]: Should this message say "appender was **not** null?"

**Comment by drewwills** [comment]
This PR looks like it does something and the code appears correct.

I'm not sure what it does or why/where I'd use it... can we get a JIRA and maybe some descriptive comments?

**Comment by ChristianMurphy** [comment]
@timlevett merge conflicts have cropped up, could you resolve them?

**Comment by apetro** [comment]
In some circumstances MyUW might redirect attempts to render a portlet via `/uPortal`'s traditional rendering pipeline to instead send the user to `AngularJS-portal`'s usage of `EXCLUSIVE` window state to render that portlet. 

It's a neat trick, and one that I think will become more relevant if/when there's more adoption of AngularJS-portal in concert with uPortal.

We're going to have to sooner or later audit the diff between MyUW-forked-uPortal and real-uPortal in the course of our eventual upgrade to uPortal 4.3 or to 5.0. I'd favor closing out this pull request un-merged for now knowing that if this feature is important it will come up again in that audit and someone will have to deal with it then.

**Comment by bjagg** [comment]
Per @apetro closing this PR.

**Pull request closed without merging**